### PR TITLE
enh: Gender as Dropdown in Attendee Form

### LINKED ISSUE #3176: Attendee Form: Make gender field a dropdown instead of textbox
**Describe the bug**

`Gender Field` is currently a textbox in `Attendee Form`.

**To Reproduce**

Steps to reproduce the behaviour:
1. Go to 'Gender Field'
2. See error

**Expected behaviour**

 Please make it a dropdown as done in `Speaker Form`
**Screenshots**

image

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Chrome
 - Version 24

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #3176

#### Short description of what this resolves:

- Made gender as a dropdown in `Attendee Form`

image

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by kushthedude** [comment]
@shreyanshdwivedi @uds5501 @prateekj117 Please Review

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Please Review

File: app/mixins/custom-form.js [hunk 1]
```diff
@@ -286,7 +286,7 @@ export default Mixin.create(MutableArray, {
       this.store.createRecord('custom-form', {
         fieldIdentifier : 'gender',
         form            : 'attendee',
-        type            : 'text',
+        type            : 'select',
```
Comment (uds5501) [inline app/mixins/custom-form.js]: @kushthedude you have to change this in server schema too and the helper functions which create the custom forms automatically.

Comment (kushthedude) [reply app/mixins/custom-form.js]: Why make change in schema ? The resulting value is getting stored in, String format .

**Comment by kushthedude** [comment]
> @kushthedude LGTM, but now you will have to write a migration file too, which takes care of all the records which were already created. Send a PR on the server for that.


@CosmicCoder96 I don't think a migration file is needed, As in this PR https://github.com/fossasia/open-event-server/pull/5909/files when the gender field type was changes it was changed in whole custom_form schema, There we didn't explicitly define the relation of gender with any object like Speaker or Attendee.

**Comment by CosmicCoder96** [comment]
Database also stores the type for each custom form entry. Refactoring here
won’t refactor the dB.

**Comment by kushthedude** [comment]
@CosmicCoder96 Server PR is merged https://github.com/fossasia/open-event-server/pull/6117

File: app/templates/components/forms/orders/order-form.hbs [hunk 2]
```diff
@@ -57,9 +57,22 @@
                 {{#if field.isLongText}}
                   {{widgets/forms/rich-text-editor value=(mut (get holder field.fieldIdentifier))
                                                    name=(if field.isRequired (concat field.fieldIdentifier '_required_' index) (concat field.fieldIdentifier '_' index))}}
+                {{else if (eq field.fieldIdentifier 'gender')}}
+                  {{#ui-dropdown class='search selection' selected=holder.gender onChange=(action (mut holder.gender)) as |execute mapper|}}
```
Comment (CosmicCoder96) [inline app/templates/components/forms/orders/order-form.hbs]: use (get holder field.fieldIdentifier) for uniformity in code.

Comment (kushthedude) [reply app/templates/components/forms/orders/order-form.hbs]: @CosmicCoder96 Did the changes

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Please Review

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Please REview

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Please REview

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Please REview

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Please REview

**Pull request merged**